### ISSUE TITLE
fulcro declares transitive dependencies as direct

### ISSUE THREAD

**Issue by thheller** [fulcrologic/fulcro#319]
With `2.8.1+` fulcro started declaring all its own transitive dependencies as direct dependencies. This confuses the dependency conflict resolution in shadow-cljs.edn and causes wrong versions to be picked.

I guess this happened as a result of moving to `deps.edn`.

For comparison check the dependencies listed under

- https://clojars.org/fulcrologic/fulcro/versions/2.8.0
- https://clojars.org/fulcrologic/fulcro/versions/2.8.1

Those dependencies should really be limited to the actual direct dependencies and `clojurescript` declared as `:provided` as before.

**Comment by awkay** [comment]
I think that was in response to clj-doc not working...don't remember. I'll see what I can do.

**Comment by awkay** [comment]
should be fixed in next release

**Comment by thheller** [comment]
I don't think this is fixed. By the looks of it the `lein-tools-deps` plugin thing just doesn't do this correctly.

When I run `lein pom` and inspect the generated `pom.xml` it still contains way too many dependencies. And although some of them are declared as `provided`, their dependencies in turn are not.

```xml
<dependency>
  <groupId>org.clojure</groupId>
  <artifactId>clojurescript</artifactId>                                                      
  <version>1.10.439</version>                                                                                                              
  <scope>provided</scope>
</dependency> 

;; this shouldn't even be here
<dependency>
  <groupId>com.google.javascript</groupId>
  <artifactId>closure-compiler-unshaded</artifactId>
  <version>v20180805</version>
</dependency>
```

**Comment by thheller** [comment]
See https://github.com/RickMoynihan/lein-tools-deps/issues/91

**Comment by awkay** [comment]
Ah, well, in f3 I've just switched to using maven directly.  I'll do that here as well.  Getting all this crap to work right (with cljdoc, etc.) is a major pain.

**Comment by awkay** [comment]
This looks better:
https://clojars.org/fulcrologic/fulcro/versions/2.8.13-SNAPSHOT

**Comment by awkay** [comment]
argh....clj -Spom doesn't support updating transitive deps, so cljdoc breaks again...

**Comment by awkay** [comment]
released in 2.8.13
